src/torznab.ts

```diff
@@ -681,3 +662,76 @@ async function makeRequests(
 		candidates: results,
 	}));
 }
+async function getAndLogIndexers(
+	enabledIndexers: Indexer[],
+	timestampDataSql,
+	mediaType: MediaType,
+	searchee: Searchee,
+): Promise<{ indexersToUse: Indexer[]; parsedMedia?: ParsedMedia }> {
+	const { excludeRecentSearch, excludeOlder } = getRuntimeConfig();
+	const mediaTypeStr = mediaType.toUpperCase();
+
+	const indexersToUseByTimestamp = enabledIndexers.filter((indexer) => {
+		const entry = timestampDataSql.find(
+			(entry) => entry.indexerId === indexer.id,
+		);
+		if (!entry) {
+			return true;
+		}
+		return (
+			(!excludeOlder || entry.firstSearched > nMsAgo(excludeOlder)) &&
+			(!excludeRecentSearch ||
+				entry.lastSearched < nMsAgo(excludeRecentSearch))
+		);
+	});
+
+	const indexersToUse = indexersToUseByTimestamp.filter((indexer) => {
+		return indexerDoesSupportMediaType(
+			mediaType,
+			JSON.parse(indexer.categories),
+		);
+	});
+	const skippingSearching = (reason: string) =>
+		`Skipped searching on indexers for ${searchee.name} ${reason} [${mediaTypeStr}]`;
+
+	let skippingIndexersCallout =
+		enabledIndexers.length > indexersToUse.length ? " (filtered by " : "";
+
+	if (enabledIndexers.length > indexersToUseByTimestamp.length) {
+		skippingIndexersCallout +=
+			indexersToUseByTimestamp.length === indexersToUse.length
+				? "timestamps)"
+				: "timestamps ";
+	}
+	if (indexersToUseByTimestamp.length > indexersToUse.length) {
+		skippingIndexersCallout +=
+			enabledIndexers.length === indexersToUseByTimestamp.length
+				? "supported category)"
+				: "and supported category)";
+	}
+	if (indexersToUse.length === 0) {
+		logger.info({
+			label: Label.TORZNAB,
+			message: `${skippingSearching(skippingIndexersCallout)}`,
+		});
+		throw new Error("SKIPPED");
```

Comment:
I don't really like the use of throw/catch for control flow like this. is there a way to achieve this behavior without throw/catch?